### ISSUE TITLE
Chunk not found error when piping gridfs file bigger than 256kb

### ISSUE THREAD

**Issue by ualtinok** [marcello3d/node-mongolian#82]
Mongo version: 2.0.2
Node vers: 0.6.8
Mongolian: 0.1.16

Using the same mongodb connection after accessing a small file, if a file bigger than 256kb in size is accessed, it throws an error like this: [Error: Chunk not found: 1/2]

``` javascript
var Db= require('mongodb').Db,
    Server= require('mongodb').Server,
    Mongolian = require('mongolian'),
    fs = require('fs');
var server = new Mongolian(
    "server1",
    "server2",
    "server3"
);
var db = server.db("fstore");

http.createServer(function(req, res) {
      var filename = getfilenamereq(req);
      gridfs.findOne(filename, function (err, file) {
            if (!err && file) {
                var stream = file.readStream();
                stream.on("error", function(err){
                    console.log(err); //THROWS  [Error: Chunk not found: 1/2]
                }).pipe(res);
            }
            else
            {
                res.writeHead(404, {'Content-Type': 'text/plain'});
                res.write('404 Not Found\n');
                res.end();
            }
      });
}).listen(88);
```

**Comment by marcello3d** [comment]
How did you insert the node? Can you make a start-to-finish test case?

**Comment by ualtinok** [comment]
The second call to findOne throws the error.
I inspected the code and found out after calling findOne second time, 
chunk's id is passed to getChunk function in gridfile on second chunk instead of file's id.

<pre><code>var Db = require('mongodb').Db,
  Server = require('mongodb').Server,
  Mongolian = require('mongolian'),
  GridStore = require('mongodb').GridStore;
var fs = require("fs");
var mongoserver = new Mongolian(
    "server1",
    "server2",
    "server3"
);
var mongoDb = mongoserver.db("ximagestore");
var gridfs = mongoDb.gridfs();

var filename = "longfile.gif";

var mf = gridfs.create({
    filename: filename,
    contentType: 'image/gif'
});

var gStream = mf.writeStream();
var rStream = fs.createReadStream(filename);
rStream.on("end", function(){
    console.log("File wrote");
}).on("error", function(err){
    throw err;
}).pipe(gStream);

var wStream1 = fs.createWriteStream(filename+".1");
var wStream2 = fs.createWriteStream(filename+".2");
gridfs.findOne(filename, function (err, file) {
    if (!err && file) {
        var stream = file.readStream();
        stream.on("error", function(err){
            console.log(err);
        }).on("end", function(){
            console.log("END");
        }).pipe(wStream1);
    }
    else
    {
        console.log("FILE NOT FOUND");
    }
});

//THIS WILL THROW CHUNK NOT FOUND ERROR
gridfs.findOne(filename, function (err, file) {
    if (!err && file) {
        var stream = file.readStream();
        stream.on("error", function(err){
            console.log(err);
        }).on("end", function(){
            console.log("END");
        }).pipe(wStream2);
    }
    else
    {
      console.log("FILE NOT FOUND");
    }
});
</code></pre>

**Comment by marcello3d** [comment]
I wonder if this is related to GH-83.

**Comment by marcello3d** [comment]
The latest master and should fix this problem. I'm adding more gridfs unit tests (and welcome contributions!).

**Comment by marcello3d** [comment]
This should be fixed in v0.1.17.
